pkg/compact/downsample/downsample.go

```diff
@@ -153,9 +155,12 @@ func Downsample(
 			for _, c := range chks {
 				ac, ok := c.Chunk.(*AggrChunk)
 				if !ok {
-					return id, errors.Errorf("expected downsampled chunk (*downsample.AggrChunk) got %T instead for series: %d", c.Chunk, postings.At())
+					// Downsampled block can erroneously contain XOR chunks, skip those
```

Comment:
Do you mind checking if it's empty XOR? This will allow us to be sure! Thanks 💪🏽

Reply:
Thanks, adjusted as suggested, please review!